packages/components/addon/components/hds/disclosure-primitive/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { schedule } from '@ember/runloop';
+
+export default class HdsDisclosurePrimitiveComponent extends Component {
+  @tracked isOpen; // notice: if in the future we need to add a "@isOpen" prop to control the status from outside (eg to have the DisclosurePrimitive opened on render) just add  "this.args.isOpen" here to initalize the variable
+  @tracked toggleRef;
+
+  @action
+  onClickToggle(event) {
+    // we store a reference to the DOM node that has the "onClickToggle" event associated with it
+    if (!this.toggleRef) {
+      this.toggleRef = event.currentTarget;
+    }
```

Comment:
This logic can go too, there's no need to store the reference on this primitive.